pyanaconda/core/util.py

```diff
@@ -1496,20 +1495,35 @@ def __setattr__(self, name, value):
         return setattr(self._object, name, value)
 
 
-def get_os_version(sysroot=""):
-    """Find version of the OS from the os-release file.
+def get_os_release_value(name, sysroot=""):
+    """Read os-release files and return a value of the specified parameter.
 
-    See os-release(5).
-
-    :param str sysroot: Where to look.
-    :return str: The version
+    :param name: a name of the parameter (for example, "VERSION_ID")
+    :param sysroot: a path to the system root
+    :return: a string with the value of None if nothing found
     """
-    for filename in ("/etc/os-release", "/usr/lib/os-release"):
+    # Match the variable assignment (for example, "VERSION_ID=").
+    name += "="
+
+    # Search all os-release files in the system root.
+    paths = ("/etc/os-release", "/usr/lib/os-release")
+
+    for path in paths:
         try:
-            with open(sysroot + filename, "r") as f:
-                data = f.read(4096)  # 4 kB should be enough for everyone!
-            return OS_RELEASE_OS_VERSION.findall(data)[0]
-        except (FileNotFoundError, IndexError):
+            with open(join_paths(sysroot, path), "r") as f:
+                for line in f:
+                    # Match the current line.
+                    if not line.startswith(name):
+                        continue
+
+                    # Get the value.
+                    value = line[len(name):]
+
+                    # Strip spaces and then quotes.
+                    value = value.strip().strip("\"'")
+                    return value
+        except FileNotFoundError:
             pass
 
+    # No value found.
```

Comment:
Maybe adding debug logging could help in some situations.

Reply:
Fixed.